environment/src/main/java/com/sequenceiq/environment/environment/validation/validators/EncryptionKeyValidator.java

```diff
@@ -0,0 +1,27 @@
+package com.sequenceiq.environment.environment.validation.validators;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+
+@Component
+public class EncryptionKeyValidator {
+
+    private static final Pattern ENCRYPTION_KEY_PATTERN = Pattern.compile("projects\\/[a-zA-Z0-9_-]{1,63}\\/" +
+            "locations\\/[a-zA-Z0-9_-]{1,63}\\/keyRings\\/[a-zA-Z0-9_-]{1,63}\\/cryptoKeys\\/[a-zA-Z0-9_-]{1,63}");
+
+    public ValidationResult validateEncryptionKey(String encryptionKey) {
+        ValidationResult.ValidationResultBuilder validationResultBuilder = ValidationResult.builder();
+        Matcher matcher = ENCRYPTION_KEY_PATTERN.matcher(encryptionKey.trim());
+        if (!matcher.matches()) {
+            validationResultBuilder.error(String.format("Expected Format: '/projects/<projectName>/locations/<location>/" +
+                    "keyRings/<KeyRing>/cryptoKeys/<Key name>'. " +
```

Comment:
nit: "keyRings/<keyRing>/cryptoKeys/<keyName>'. "

camel case consistency. Later it is referred to as keyName.